schema/gen/go/testMaps_test.go

```diff
@@ -58,7 +59,7 @@ func TestMapsContainingMaybe(t *testing.T) {
 					ma.AssembleEntry("one").AssignString("1")
 					ma.AssembleEntry("two").AssignString("2")
 				})
-				Wish(t, n, ShouldEqual, nr)
+				Wish(t, bind.Unwrap(n), ShouldEqual, bind.Unwrap(nr))
```

Comment:
Okay, so I realize this was probably a "fasted way to adapt these tests for reuse" thing, but it's a pretty goodsized wat.

I'd like to have these tests not needing to import the bind package.  To do so is just a really weird shape for the import graph.

Reply:
In fact, it would be pretty nice to remove these from the `schema/gen` prefix, since we're now seeing they're definitely more broadly applicable than that (rather than just hopefully so :sweat_smile: ).

I don't know how much work that is / if we want to couple it to this diff.  If we _could_ get these extracted in one PR, and then use it for the new bind features in an immediately subsequent PR, though, that'd be super dreamy.

Reply:
As for the DeepEqual itself, I don't think it's okay for the tests to assume that will work on any ipld.Node implementation. Maybe the "engine" passed to the nodetest package should be an interface, and another method could be "get me the underlying comparable value of this node", or "tell me if these two nodes are equal".

Another option is to declare a semi-standard "Unwrap" optional method on Node types. I have a TODO for that on my code, but I felt like it was too early to bring that up.

Reply:
Done; removed all the test changes from this PR, they'll come in a follow-up PR.

Reply:
Yes... I think we could also create a generic `Equals` function for `ipld.Node` (traversals of them are well defined after all) and that would be a generally reasonable and useful thing to export even beyond tests.  Or, even more powerfully (and helpful for debugging), a `Diff` feature, though that's more work.

We could also go further and make [`Checker`](https://pkg.go.dev/github.com/frankban/quicktest#Checker) implementations (or, [`Checker`](https://pkg.go.dev/github.com/warpfork/go-wish#Checker), depending on how quickly we migrate testing libraries) which use that feature.

Reply:
Oh, right, we could have our own `ipld.DeepEqual` which would do the same kind of nested comparison that `reflect.DeepEqual` does, but at the IPLD data model layer instead of Go reflection layer.

Reply:
Exactly :D